main/tasks/stratum_task.c

```diff
@@ -48,313 +44,335 @@ struct timeval tcp_rcv_timeout = {
-                    stratum_api_v1_message.method == STRATUM_RESULT_VERSION_MASK) {
-                // 1fffe000
-                ESP_LOGI(TAG, "Set version mask: %08lx", stratum_api_v1_message.version_mask);
-                GLOBAL_STATE->version_mask = stratum_api_v1_message.version_mask;
-                GLOBAL_STATE->new_stratum_version_rolling_msg = true;
-            } else if (stratum_api_v1_message.method == STRATUM_RESULT_SUBSCRIBE) {
-                GLOBAL_STATE->extranonce_str = stratum_api_v1_message.extranonce_str;
-                GLOBAL_STATE->extranonce_2_len = stratum_api_v1_message.extranonce_2_len;
-            } else if (stratum_api_v1_message.method == CLIENT_RECONNECT) {
-                ESP_LOGE(TAG, "Pool requested client reconnect...");
-                stratum_close_connection(GLOBAL_STATE);
-                break;
-            } else if (stratum_api_v1_message.method == STRATUM_RESULT) {
-                if (stratum_api_v1_message.response_success) {
-                    ESP_LOGI(TAG, "message result accepted");
-                    SYSTEM_notify_accepted_share(GLOBAL_STATE);
-                } else {
-                    ESP_LOGW(TAG, "message result rejected: %s", stratum_api_v1_message.error_str ? stratum_api_v1_message.error_str : "unknown");
-                    SYSTEM_notify_rejected_share(GLOBAL_STATE);
-                }
-            } else if (stratum_api_v1_message.method == STRATUM_RESULT_SETUP) {
-                // Reset retry attempts after successfully receiving data.
-                retry_attempts = 0;
-                if (stratum_api_v1_message.response_success) {
-                    ESP_LOGI(TAG, "setup message accepted");
-                } else {
-                    ESP_LOGE(TAG, "setup message rejected: %s", stratum_api_v1_message.error_str ? stratum_api_v1_message.error_str : "unknown");
-                }
+            connection->message->mining_notification->difficulty = connection->stratum_difficulty;
+            connection->message->mining_notification->connection_id = connection->id;
+            queue_enqueue(&connection->stratum_queue, connection->message->mining_notification);
+        }
+        else if (connection->message->method == MINING_SET_DIFFICULTY)
+        {
+            if (connection->stratum_difficulty != connection->message->new_difficulty)
+            {
+                connection->stratum_difficulty = connection->message->new_difficulty;
+                ESP_LOGI(POOL_TAG, "Set stratum difficulty: %ld", connection->stratum_difficulty);
+            }
+        }
+        else if (connection->message->method == MINING_SET_VERSION_MASK ||
+                connection->message->method == STRATUM_RESULT_VERSION_MASK)
+        {
+            ESP_LOGI(POOL_TAG, "Set version mask: %08lx", connection->message->version_mask);
+            connection->version_mask = connection->message->version_mask;
```

Comment:
For reviewer's clarity, please keep large-ish reformattings out of a PR. This throws off the diff viewer of GitHub and makes it more difficult to follow what's actually changed. Here the side by side diff is completely misaligned.

Reply:
Yep fully agree. In this case the code went through many iterations and the PR originally looked very different. Will keep in mind for the future.

Reply:
It also makes conflicts with other PRs less likely, that would be the biggest benefit.